File: ContainerWithMostWater/ContainerWithMostWater.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        left = 0
        right = len(height) - 1
        result = 0
        while i < j:
        	tmp = (j - i) * min(height[left], height[right])
        	if height[left] > height[right]:
        		right -= 1
        	else:
        		left += 1
        return result

File: ContainerWithMostWater/ContainerWithMostWater.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int result = 0;
        // if(height.size() < 2)
        // 	return result;
        int i = 0, j = height.size() - 1, tmp;
        while(i < j) {
        	tmp = (j - i) * min(height[i], height[j]);
        	result = max(tmp, result);
        	if(height[i] < height[j])
        		i++;
        	else
        		j--;
        }
        return result;
    }
};